core/src/main/scala/org/apache/spark/deploy/security/HadoopAccessManager.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.security
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+
+/**
+ * Methods in [[HadoopAccessManager]] return scheduler-specific information related to how Hadoop
+ * delegation tokens should be fetched.
+ */
+private[spark] trait HadoopAccessManager {
+
+  /** The user allowed to renew delegation tokens */
+  def getTokenRenewer: String
+
+  /** The renewal interval, or [[None]] if the token shouldn't be renewed */
+  def getTokenRenewalInterval: Option[Long]
```

Comment:
That's a good point.  They don't.  I have it factored out like this because the code which computes it depends on `spark.yarn.principal`, which of course should be specific to YARN, but once Mesos keytab/principal support gets merged, that variable will likely be deprecated and replaced with `spark.kerberos.principal` or something.

I'll go ahead and remove this function from this trait and put it back in `HadoopFSCredentialProvider`.  It will use `spark.yarn.principal` until we rename it later.